### PR TITLE
Pluggable anchors select itests

### PR DESCRIPTION
**Pull request by halseth**
This PR adds integration tests for a select set of integration tests with the channel type set to anchor:

- basic funding flow
- channel force closure
- test multi-hop htlc local force close immediate expiry
- test multi-hop htlc receiver chain claim
- test multi-hop local force close on-chain htlc timeout
- test multi-hop remote force close on-chain htlc timeout
- test multi-hop htlc local chain claim
- test multi-hop htlc remote chain claim

These are selected since they together test establishing channels of this new type in combination with the previous channel types, sweeping of commitment outputs, and HTLC sweep scenarios.

The goal is to later enable the whole integration test suite for this channel type, but it requires changes to most notably the channel backup format and watchtower.

Builds on #3821

### CODE REVIEW

**Comment by joostjager** [comment]
`TestLightningNetworkDaemon/basic_funding_flow/carol_commit=anchors,dave_commit=anchors` fails

File: rpcserver.go [hunk 1]
```diff
@@ -2556,6 +2556,14 @@ func (r *rpcServer) ChannelBalance(ctx context.Context,
 	var balance btcutil.Amount
 	for _, channel := range openChannels {
 		balance += channel.LocalCommitment.LocalBalance.ToSatoshis()
+
+		// To be consistent with balance shown across channels
+		// regardless of channel type, add any anchor fees we are
+		// paying for the two anchors back to our balance if we are the
+		// initiator.
+		if channel.IsInitiator {
```
Comment (halseth) [inline rpcserver.go]: report real balance, anchors as separate field

File: lnrpc/rpc.proto [hunk 2]
```diff
@@ -1356,6 +1356,14 @@ message Channel {
     */
     bool static_remote_key = 22 [json_name = "static_remote_key"];
 
+
+    /**
+    If non-zero, then this channel uses a commitment format that has anchor
+    outputs on the commitments, allowing fee bumping after a force close
+    transaction has been broadcast.
+    */
+    int64 anchor_output_size = 26 [json_name = "anchor_output_size"];
```
Comment (halseth) [inline lnrpc/rpc.proto]: move to firt PR

File: lnrpc/rpc.proto [hunk 3]
```diff
@@ -1356,6 +1356,14 @@ message Channel {
     */
     bool static_remote_key = 22 [json_name = "static_remote_key"];
 
+
+    /**
+    If non-zero, then this channel uses a commitment format that has anchor
```
Comment (halseth) [inline lnrpc/rpc.proto]: take into account one or two anchors on commitment?

File: lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go [hunk 4]
```diff
@@ -498,3 +498,76 @@ func checkPaymentStatus(ctxt context.Context, node *lntest.HarnessNode,
 
 	return nil
 }
+
+func createThreeHopNetwork(t *harnessTest, net *lntest.NetworkHarness,
```
Comment (halseth) [inline lntest/itest/lnd_multi-hop_htlc_local_chain_claim_test.go]: move in separate (trivial) PR?

**Comment by Roasbeef** [comment]
As brought up in the other PR, we should also expand the tests for breaches as well.

**Comment by joostjager** [comment]
Can be rebased on top of #4103

**Comment by joostjager** [comment]
Superseded by #4073

**Pull request closed without merging**